src/Core/Content/Product/SalesChannel/Review/ProductReviewSaveRoute.php

```diff
@@ -120,54 +107,38 @@ private function validate(DataBag $data, Context $context): void
     {
         $definition = new DataValidationDefinition('product.create_rating');
 
-        $definition->add('name', new NotBlank());
         $definition->add('title', new NotBlank(), new Length(['min' => 5]));
         $definition->add('content', new NotBlank(), new Length(['min' => 40]));
 
         $definition->add('points', new GreaterThanOrEqual(1), new LessThanOrEqual(5));
 
+        $reviewValidationCriteria = new Criteria();
         if ($data->get('id')) {
-            $criteria = new Criteria();
-            $criteria->addFilter(new EqualsFilter('customerId', $data->get('customerId')));
-            $criteria->addFilter(new EqualsFilter('id', $data->get('id')));
+            $reviewValidationCriteria->addFilter(new EqualsFilter('customerId', $data->get('customerId')));
 
             $definition->add('id', new EntityExists([
                 'entity' => 'product_review',
                 'context' => $context,
-                'criteria' => $criteria,
+                'criteria' => $reviewValidationCriteria,
             ]));
         } else {
-            $criteria = new Criteria();
-            $criteria->addFilter(new EqualsFilter('customerId', $data->get('customerId')));
-            $criteria->addFilter(new EqualsFilter('productId', $data->get('productId')));
+            $reviewValidationCriteria->addFilter(new EqualsFilter('productId', $data->get('productId')));
 
             $definition->add('customerId', new EntityNotExists([
                 'entity' => 'product_review',
                 'context' => $context,
-                'criteria' => $criteria,
+                'criteria' => $reviewValidationCriteria,
+                'primaryProperty' => 'customerId',
```

Comment:
We can not add \`primaryProperty\` here. That should follow the type of the $option at EntityNotExists.php::__construct()

So please update the type-hint as well